### PR TITLE
Add STAR aligner

### PR DESCRIPTION
**Pull request by edmundmiller**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/nascent/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/nascent _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/igenomes.config [hunk 1]
```diff
@@ -37,11 +37,8 @@ params {
             blacklist   = "${projectDir}/assets/blacklists/hg38-blacklist.bed"
         }
         'CHM13' {
-            fasta       = "${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/WholeGenomeFasta/genome.fa"
-            bwa         = "${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/"
-            bwamem2     = "${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/"
-            gtf         = "${params.igenomes_base}/Homo_sapiens/NCBI/CHM13/Annotation/Genes/genes.gtf"
-            gff         = "ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/009/914/755/GCF_009914755.1_T2T-CHM13v2.0/GCF_009914755.1_T2T-CHM13v2.0_genomic.gff.gz"
+            fasta       = "https://s3-us-west-2.amazonaws.com/human-pangenomics/T2T/CHM13/assemblies/analysis_set/chm13v2.0.fa.gz"
+            gff         = "https://s3-us-west-2.amazonaws.com/human-pangenomics/T2T/CHM13/assemblies/annotation/chm13v2.0_RefSeq_Liftoff_v5.1.gff3.gz"
```
Comment (maxulysse) [inline conf/igenomes.config:41]: Can you parametrize that?
So that it could be use easily with local data

File: workflows/nascent.nf [hunk 2]
```diff
@@ -14,7 +14,11 @@ include { COVERAGE_GRAPHS } from '../subworkflows/local/coverage_graphs.nf'
 include { TRANSCRIPT_INDENTIFICATION } from '../subworkflows/local/transcript_identification.nf'
 
 include { FASTP } from '../modules/nf-core/fastp/main'
-include { UNTAR as UNTAR_HISAT2_INDEX } from '../modules/nf-core/untar/main'
+include {
+    UNTAR as UNTAR_HISAT2_INDEX
+    UNTAR as UNTAR_STAR_INDEX
+} from '../modules/nf-core/untar/main'
```
Comment (maxulysse) [inline workflows/nascent.nf:20]: I'd rather do 2 include statement, it reads nicer IMO

File: main.nf [hunk 3]
```diff
@@ -38,6 +38,7 @@ params.bwamem2_index = getGenomeAttribute('bwamem2')
 params.dragmap = getGenomeAttribute('dragmap')
 params.bowtie2_index = getGenomeAttribute('bowtie2')
 params.hisat2_index = getGenomeAttribute('hisat2')
+params.star_index = null
```
[the remark this answers is not in the document]
Comment (edmundmiller) [reply main.nf:41]: Nope, it's not up-to-date, and I wanted to use the latest version of STAR for benchmarking 😞 

I have little motivation to hack around the current aws-igenomes quirks, I'd rather spend that time on nf-core/references 😉

File: workflows/nascent.nf (hunk 2 above)
Comment (edmundmiller) [reply workflows/nascent.nf:20]: Talk to me when someone writes a formatter 😛

Comment (edmundmiller) [reply workflows/nascent.nf:20]: I like the split includes, but I'll adhere to a style guide once there's a formatter that handles it for me.

**Pull request merged**